zerver/views/__init__.py

```diff
@@ -232,7 +236,10 @@ def accounts_register(request):
 
         if first_in_realm:
             do_change_is_admin(user_profile, True)
-            return HttpResponseRedirect(reverse('zerver.views.initial_invite_page'))
+            invite_url = reverse('zerver.views.initial_invite_page')
+            if(realm_creation and settings.REALMS_HAVE_SUBDOMAINS):
```

Comment:
You have a few more of these; I added this to the linter so it'll catch these when you next rebase.